PVTPackage/source/MultiphaseSystem/PhaseSplitModel/DeadOilFlash.cpp

```diff
@@ -62,22 +59,24 @@ DeadOilFlash::DeadOilFlash( std::vector< std::vector< double > > const & PVDO,
   :
   m_oilPhaseModel( pvt::PHASE_TYPE::OIL, PVDO, oilSurfaceMassDensity, oilSurfaceMolecularWeight )
 {
-  BlackOil_WaterModel * ptr = new BlackOil_WaterModel( PVTW, waterSurfaceMassDensity, waterSurfaceMolecularWeight );
-  m_waterPhaseModel = std::unique_ptr< BlackOil_WaterModel >( ptr );    
+  m_waterPhaseModel = std::make_unique< BlackOil_WaterModel >( PVTW, waterSurfaceMassDensity, waterSurfaceMolecularWeight );
 }
   
 DeadOil_PhaseModel const & DeadOilFlash::getOilPhaseModel() const
 {
   return m_oilPhaseModel;
 }
 
+  
 DeadOil_PhaseModel const & DeadOilFlash::getGasPhaseModel() const
 {
+  // Warning! the result returned by this function will be void for a two-phase oil-water model
```

Comment:
Argh, this should be better as a `doxygen` comment: it's for the dev who uses this class from the outside.
You should use some `@brief` here, sorry  🤷 

Also, is it _this function will be void for_ or _this function **may** be void for_?
One of those two must not be zero. Am I right?

Reply:
Sorry I missed that point. I added a Doxygen comment for each of these functions.

Yes you are right. The function  `getGasPhaseModel` will always be void for a two-phase oil-water model (no gas), and the function `getWaterPhaseModel` will always be void for a two-phase oil-gas model (no water). For three-phase flow, none of them will be void.